src/mainboard/protectli/vault_adl_p/die.c

```diff
@@ -0,0 +1,62 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+
+#include <arch/io.h>
+#include <console/console.h>
+#include <soc/gpio.h>
+#include <delay.h>
+#include <gpio.h>
+
+#define ITE_GPIO_BASE		0xa00
+#define ITE_GPIO_PIN(x)		(1 << ((x) % 10))
+#define ITE_GPIO_SET(x)		(((x) / 10) - 1)
+#define ITE_GPIO_IO_ADDR(x)	(ITE_GPIO_BASE + ITE_GPIO_SET(x))
+
+static void do_beep(uint32_t frequency, uint32_t duration_msec)
```

Comment:
It seems we have two implementations of do_beep(), is it necessary?

Reply:
It is the same implementation.

Necessary no. But making it a function to be called across modules is also not much beneficial. I would have to create a new header file `.h` and new source file `.c` just to remove couple lines of code in one of two places.

Reply:
> I would have to create a new header file .h and new source file .c

Isn't that a standard way of doing things?

Do you think it would be pointed out in upstream review as well? If yes, why not change it? If no, let's leave it like this.

Reply:
> Isn't that a standard way of doing things?

For coreboot I suspect it would most likely have to be a separate file in board directory.

> Do you think it would be pointed out in upstream review as well?

There were situations where such solutions were not deemed as much beneficial.

Reply:
@macpijan I think we may close this one. I will address review comments on gerrit to this part, if any of those appear.

Reply:
Ok, let's follow your best judgement.